tests/scripts/validate_modified_files.sh

```diff
@@ -9,26 +9,31 @@ MOD_ERR="changes found by mod.check. You may need to run make clean"
 CRD_ERR="changes found by 'make crds'. please run 'make crds' locally and update your PR"
 BUILD_ERR="changes found by make build', please commit your go.sum or other changed files"
 HELM_ERR="changes found by 'make gen-rbac'. please run 'make gen-rbac' locally and update your PR"
+DOCS_ERR="changes found by 'make docs'. please run 'make docs' locally and update your PR"
+
 
 #############
 # FUNCTIONS #
 #############
 function validate(){
-  git=$(git status --porcelain)
-  for file in $git; do
-    if [ -n "$file" ]; then
-      echo "$1"
-      echo "$git"
-      git diff
+  ERR=$1
+  git diff --exit-code
+  ret=$?
+  if [ "$ret" != "0" ]; then
```

Comment:
have you tested this,

Any easy would be updating the docs in the second commit and not running `make check.docs`

Reply:
if you have tested it then its fine,
I was just suggesting to add a second commit with some doc change and not running `make cmd`

Reply:
Following the conversation around `set -ex`, let's not change this. The script is working today, and from a code/project maintenance perspective, this seems like unnecessary risk of breaking corner cases when running the script. We already have enough CI instability, and I don't want to risk more over scripting preferences.

Reply:
@BlaineEXE  wrote:
> Following the conversation around `set -ex`, let's not change this. The script is working today, and from a code/project maintenance perspective, this seems like unnecessary risk of breaking corner cases when running the script. We already have enough CI instability, and I don't want to risk more over scripting preferences.

understood, and changed.